### PR TITLE
Init Elastisearch spec asciidoc

### PR DESCRIPTION
**Pull request by thbkrkr**
- Draft intro
- Node configuration
- HTTP settings & TLS SANs
- Custom HTTP certificate

Parts of #1031.

### CODE REVIEW

File: docs/elasticsearch-spec.asciidoc [hunk 1]
```diff
@@ -0,0 +1,103 @@
+[id="{p}-elasticsearch-specification"]
+== Elasticsearch Specification
+
+There are a number of settings which need to be considered before going into production related to Elasticsearch but also to Kubernetes.
```
Comment (sebgl) [inline docs/elasticsearch-spec.asciidoc]: I think this sentence sounds a bit strange, @alaudazzi what do you think?

File: docs/elasticsearch-spec.asciidoc [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+- Change budget (maxUnavailable, maxSurge)
+
+[id="{p}-node-configuration"]
+=== Node configuration
+
+Any setting that can be configured in the `elasticsearch.yml` configuration file can be defined for each topology of nodes in the `spec.nodes[?].config` section.
+
+[source,yaml]
+----
+kind: Elasticsearch
```
Comment (sebgl) [inline docs/elasticsearch-spec.asciidoc]: I think we should either include `Kind` + `apiVersion` + `metadata`, either none of those but only the `spec`. An in-between choice can be confusing.

File: docs/elasticsearch-spec.asciidoc (hunk 1 above)
Comment (thbkrkr) [reply docs/elasticsearch-spec.asciidoc]: I agree it's not great. I tried to write a draft of introduction.
This comes from https://www.elastic.co/guide/en/elasticsearch/reference/current/important-settings.html :)
We agreed with @alaudazzi that she will make a global pass when we have covered all the settings.

File: docs/elasticsearch-spec.asciidoc (hunk 2 above)
Comment (thbkrkr) [reply docs/elasticsearch-spec.asciidoc]: I forget to remove it.
This is because at the beginning I documented ES and Kibana.

**Pull request merged**